pkg/controller/utils/utils.go

```diff
@@ -615,3 +615,27 @@ func GetElasticsearch(ctx context.Context, c client.Client) (*esv1.Elasticsearch
 	}
 	return &es, nil
 }
+
+func IsNodeLocalDNSAvailable(ctx context.Context, cli client.Client) (bool, error) {
+	ds := &v1.DaemonSet{}
+
+	err := cli.Get(ctx, client.ObjectKey{Namespace: "kube-system", Name: "node-local-dns"}, ds)
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+// AddNodeLocalDNSWatch creates a watch on the node local dns.
+func AddNodeLocalDNSWatch(c controller.Controller) error {
+	ds := &v1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "kube-system",
+			Labels: map[string]string{
+				"k8s-app": "node-local-dns",
+			},
+		},
+	}
+	return c.Watch(&source.Kind{Type: ds}, &handler.EnqueueRequestForObject{}, createPredicateForObject(ds))
```

Comment:
I think the only thing that `source.Kind` cares about is the underlying object kind, I don't think it references the namespace or labels, i.e. I think the first argument should be `&v1.Daemonset{}`

Also, when passing this to `createPredicateForObject`, the predicate is constructed based on name and namespace. I believe since name is omitted, this will match all Daemonsets in kube-system